backend/src/main/scala/cromwell/backend/package.scala

```diff
@@ -0,0 +1,40 @@
+package cromwell
+
+import com.typesafe.config.Config
+import cromwell.core.{WorkflowId, WorkflowOptions}
+import wdl4s._
+import wdl4s.values.WdlValue
+
+package object backend {
+
+  /**
+    * For uniquely identifying a job which has been or will be sent to the backend.
+    */
+  case class BackendJobDescriptorKey (callFqn: String, index: Option[Int], attempt: Int)
+
+  /**
+    * For uniquely identifying a symbol in the symbol map
+    */
+  case class SymbolKey(fullyQualifiedName: FullyQualifiedName, index: Option[Int])
+
+  type BackendSymbolMap = Map[SymbolKey, WdlValue]
+
+  /**
+    * For passing to a BackendWorkflowActor for job execution or recovery
+    */
+  //case class BackendJobDescriptor(key: BackendJobDescriptorKey, call: Call, locallyQualifiedInputs: CallInputs)
+  case class BackendJobDescriptor(key: BackendJobDescriptorKey, call: Call, symbolMap: BackendSymbolMap)
```

Comment:
Yeah I agree a BE implementer probably doesn't care about implementing backend engine functions themselves... but for now it's in here... :)